### PR TITLE
internal ref replaced by uuid

### LINKED ISSUE #5666: Usage: Internal DB reference in description of VM snapshot on primary
see cmk results below

```
(localcloud) SBCM5> > list usagerecords type=27 startdate=2021-11-03 enddate=2021-11-03
{
  "count": 82,
  "usagerecord": [
    {
      "account": "admin",
      "accountid": "0e3e52ed-03db-11ec-8879-1e0079000360",
      "description": "VMSnapshot on primary storage usage Id: 1 (2790057b-5a2f-4f90-ae5b-021573d0943d)  for VM L2 (7a25841b-cfac-49ca-90eb-0cb187ba3414) with size (8.00 GB) 8589934592",
      "domain": "ROOT",
      "domainid": "e5ce2438-03da-11ec-8879-1e0079000360",
      "enddate": "2021-11-03'T'17:11:53+00:00",
      "name": "i-2-1000-VM_VS_20211103170813",
      "rawusage": "0.058435",
      "size": 8589934592,
      "startdate": "2021-11-03'T'17:08:23+00:00",
      "tags": [],
      "usage": "0.058435 Hrs",
      "usageid": "2790057b-5a2f-4f90-ae5b-021573d0943d",
      "usagetype": 27,
      "virtualmachineid": "7a25841b-cfac-49ca-90eb-0cb187ba3414",
      "zoneid": "59260b3d-f587-4a8c-8c58-182ae3f01679"
    },

...
```

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
Usage
~~~

##### CLOUDSTACK VERSION

~~~
4.16.0.0
~~~

##### STEPS TO REPRODUCE

~~~
1. change global settings for usage server , restart cloudstack management and usage server

enable.usage.server = true
usage.stats.job.aggregation.range = 5

2. create vm;
3. create vm snapshot
4. wait for 5-10 minutes, check usagerecord by cmk or cloud_usage.cloud_usage table (usage type=25).
~~~

##### EXPECTED RESULTS

~~~
the `description` in cmk result does not contain the usage id.


usage Id (2790057b-5a2f-4f90-ae5b-021573d0943d)

~~~

##### ACTUAL RESULTS

~~~
```
usage Id: 1 (2790057b-5a2f-4f90-ae5b-021573d0943d)
```
~~~

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR ...
Fixes #5666

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [x] Trivial

### CODE REVIEW

**Review by weizhouapache — APPROVED** [review]
@DaanHoogland 
have you checked other usage records ?

**Review by nvazquez — APPROVED** [review]

File: server/src/main/java/com/cloud/api/ApiResponseHelper.java [hunk 1]
```diff
@@ -3991,7 +3991,7 @@ public UsageRecordResponse createUsageResponse(Usage usageRecord, Map<String, Se
                 final StringBuilder builder = new StringBuilder();
                 builder.append("VMSnapshot on primary storage usage");
                 if (vmSnapshotVO != null) {
-                    builder.append(" Id: ").append(vmSnapshotVO.getId()).append(" (").append(vmSnapshotVO.getUuid()).append(") ");
+                    builder.append(" Id: ").append(vmSnapshotVO.getUuid()).append(" (").append(vmSnapshotVO.getUuid()).append(") ");
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/api/ApiResponseHelper.java:3994]: `Id: <uuid>` should be ok. no need to keep uuid twice `Id: <uuid> (<uuid>)`

Comment (harikrishna-patnala) [reply server/src/main/java/com/cloud/api/ApiResponseHelper.java:3994]: we may need the same change at line:3951 https://github.com/apache/cloudstack/blob/56ef349b0e49d00d21c63d2c2ef7bee6669e82a2/server/src/main/java/com/cloud/api/ApiResponseHelper.java#L3951

**Comment by DaanHoogland** [comment]
> @DaanHoogland have you checked other usage records ?

@weizhouapache yes, see my comment on th issue page: https://github.com/apache/cloudstack/issues/5666#issuecomment-983525640

**Comment by DaanHoogland** [comment]
@sureshanaparti @harikrishna-patnala (cc @weizhouapache ) all (both) IDs are now removed from the usagerecords details.

File: server/src/main/java/com/cloud/api/ApiResponseHelper.java [hunk 2]
```diff
@@ -3948,7 +3948,7 @@ public UsageRecordResponse createUsageResponse(Usage usageRecord, Map<String, Se
                 final StringBuilder builder = new StringBuilder();
                 builder.append("VMSnapshot usage");
                 if (vmSnapshotVO != null) {
-                    builder.append(" Id: ").append(vmSnapshotVO.getId()).append(" (").append(vmSnapshotVO.getUuid()).append(") ");
+                    builder.append(" Id: ").append(vmSnapshotVO.getUuid());
```
Comment (weizhouapache) [inline server/src/main/java/com/cloud/api/ApiResponseHelper.java:3951]: @DaanHoogland 
`.append(") ");` ?

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/api/ApiResponseHelper.java (hunk 2 above)
Comment (DaanHoogland) [reply server/src/main/java/com/cloud/api/ApiResponseHelper.java:3951]: what do you mean @weizhouapache ?

Comment (weizhouapache) [reply server/src/main/java/com/cloud/api/ApiResponseHelper.java:3951]: @DaanHoogland 
line 3994 has `append(") ");` at the end, should it be removed ?

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/api/ApiResponseHelper.java:3951]: ah good catch, thanks

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by GutoVeronezi — APPROVED** [review]
CLGTM

**Review by sureshanaparti — APPROVED** [review]

**Comment by sureshanaparti** [comment]
The test failure above is not related to this PR changes. Merging based on the LGTMs and tests.

**Pull request merged**